plugin/keyring_vault/vault_curl.h

```diff
@@ -0,0 +1,52 @@
+#ifndef MYSQL_VAULT_CURL_H
+#define MYSQL_VAULT_CURL_H
+
+#include <my_global.h>
+#include <curl/curl.h>
```

Comment:
I do not like the idea of not including curl/curl.h in header file. I does not limit the compilation time greatly and over complicates things for no good reason.